webui/react/src/shared/utils/color.ts

```diff
@@ -175,3 +175,17 @@ export const str2rgba = (str: string): RgbaColor => {
 
   return { a: 0.0, b: 0, g: 0, r: 0 };
 };
+
+/** check if r g b for a single color are equal within a tolerance. color is almost monochrome */
+export const isMonochrome = (rgba: RgbaColor, tolerance = 0): boolean => {
+  return (Math.abs(rgba.r - rgba.g) < tolerance && Math.abs(rgba.g - rgba.b) < tolerance);
+};
+
+/** calculate max color distance between two rgba colors */
```

Comment:
Ah color space... ok so this can get long winded, so I'll try to keep it short... So the color space we usually care about is using the [CIELAB Color Space](https://en.wikipedia.org/wiki/CIELAB_color_space). There are other variations of it but this is the granddaddy of them. What is unique about this one is that it's a color space designed around the human vision and what we perceive as color deltas.

So in order to calculate the color space, we want to compare the CIELAB colors. The formula is this:
```
ΔE = √ (L*₁-L*₂)² + (a*₁-a*₂)² + (b*₁-b*₂)².
```
But to do this we need the CIELAB colors, which we'll need to convert from RGB colors. But there isn't a simple straight forward conversion directly but rather you can convert RGB to XYZ CIE Color space color, which then convert over to CIELAB colors. So on [this page](http://www.easyrgb.com/en/math.php) contains two sets of formulas we'll want.
1. XYZ <===> Standard-RGB
2. XYZ <===> CEI-L*ab
I know it's a lot more work but this will produce much better human distinguishable colors...

Reply:
thanks for all the details. let me read through the articles. I can take a stab at this

Reply:
there is room to improve and prove the correctness on how we compute CIELAB color and perceptual distance between colors but I think we should leave those as future work. namely bringing the formulas from easyrgb

Reply:
https://determinedai.atlassian.net/browse/DET-7974